### ISSUE TITLE
Failing to pre-train Bert with TPU V2 8 Cores, Google Cloud Storage, 30 GB RAM and TF 1.11.0

### ISSUE THREAD

**Issue by TalhaSami** [google-research/bert#899]
Hi,

I am getting the below error when trying to pre-train BERT with TPU V2 8 Cores, 30 GB RAM, Google Cloud Storage and TF 1.11.0

INFO:tensorflow:Enqueue next (1) batch(es) of data to infeed.
INFO:tensorflow:Dequeue next (1) batch(es) of data from outfeed.
**INFO:tensorflow:Error recorded from infeed: End of sequence**
         [[{{node input_pipeline_task0/while/IteratorGetNext}} = IteratorGetNext[_class=["loc:@input_pipeline_ta
sk0/while/InfeedQueue/enqueue/0"], output_shapes=[[1,128], [1,128], [1,5], [1,5], [1,5], [1,1], [1,128], [1,128]
], output_types=[DT_INT32, DT_INT32, DT_INT32, DT_INT32, DT_FLOAT, DT_INT32, DT_INT32, DT_INT32], _device="/job:
worker/replica:0/task:0/device:CPU:0"](input_pipeline_task0/while/IteratorGetNext/Enter, ^input_pipeline_task0/w
hile/Identity)]]

**Caused by op u'input_pipeline_task0/while/IteratorGetNext', defined at:**
  File "run_pretraining.py", line 552, in <module>
    tf.app.run()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "run_pretraining.py", line 525, in main
    estimator.train(input_fn=train_input_fn, max_steps=FLAGS.num_train_steps)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2394, i
n train
    saving_listeners=saving_listeners
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py", line 356, in train
    loss = self._train_model(input_fn, hooks, saving_listeners)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py", line 1181, in _train_m
odel
    return self._train_model_default(input_fn, hooks, saving_listeners)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py", line 1211, in _train_m
odel_default
    features, labels, model_fn_lib.ModeKeys.TRAIN, self.config)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2186, i
n _call_model_fn
    features, labels, mode, config)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py", line 1169, in _call_mo
del_fn
....
File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/data/ops/iterator_ops.py", line 420, in get_next
    name=name)), self._output_types,
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gen_dataset_ops.py", line 2069, in iterator_get_next
    output_shapes=output_shapes, name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 3272, in create_op
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 1768, in __init__
    self._traceback = tf_stack.extract_stack()

OutOfRangeError (see above for traceback): End of sequence
         [[{{node input_pipeline_task0/while/IteratorGetNext}} = IteratorGetNext[_class=["loc:@input_pipeline_task0/while/InfeedQueue/enqueue/0"], output_shapes=[[1,128], [1,128], [1,5], [1,5], [1,5], [1,1], [1,128], [1,128]], output_types=[DT_INT32, DT_INT32, DT_INT32, DT_INT32, DT_FLOAT, DT_INT32, DT_INT32, DT_INT32], _device="/job:worker/replica:0/task:0/device:CPU:0"](input_pipeline_task0/while/IteratorGetNext/Enter, ^input_pipeline_task0/while/Identity)]]

**INFO:tensorflow:An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: Socket closed**
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from gs://bert-project-bucket/model.ckpt-0

I am trying to pretrain the lightest-version and very small batch-sizes, but still getting the above errors.

My issue seems similar to https://github.com/google-research/bert/issues/486, but someone commented there that their issue was resolved by using small batch_sizes (while mine isn't).

I was not getting this when I tried to train BERT on my CPU, but then it would have taken very long for the training to complete.

My other params are as follows:

--max_seq_length=128
--max_predictions_per_seq=20
--num_train_steps=20
--num_warmup_steps=10
--learning_rate=2e-5

and the bert_config is the standard for BERT-Base Uncased:

"attention_probs_dropout_prob": 0.1,
"hidden_act": "gelu",
"hidden_dropout_prob": 0.1,
"hidden_size": 768,
"initializer_range": 0.02,
"intermediate_size": 3072,
"max_position_embeddings": 512,
"num_attention_heads": 12,
"num_hidden_layers": 12,
"type_vocab_size": 2,
"vocab_size": 30522

**Comment by kwonmha** [comment]
Hi, I got the same error.
Granting owner permission of my storage bucket to my tpu service account solved the problem.
Try this :
 `gsutil acl -r ch -u YOUR_TPU_SERVICE_ACCOUNT:OWNER gs://YOUR_BUCKET`

You can find your tpu service account if you click your tpu instance.
It looks like service-{your-project-id}@cloud-tpu.iam.gserviceaccount.com

You may need to add -r option to give permission of whole subdirectories, files.

**Comment by TalhaSami** [comment]
Thanks kwonmha, but the error still happens in my case even after I have granted myself the owner permission of my storage bucket (and all subdirectories).

**Comment by soares-f** [comment]
I am having the same issue with V3-8

**Comment by mananeau** [comment]
> Thanks kwonmha, but the error still happens in my case even after I have granted myself the owner permission of my storage bucket (and all subdirectories).

I also got the same error with V3-8, even after granting both the TPU service and myself the owner permission. Any solutions?

**Comment by mananeau** [comment]
I personnally solved it by having the --input_file pointing to files in my GCP bucket. In the past, only the --output_dir was pointing to my GCP bucket and I was getting this error. Hope this helps!
